### ISSUE TITLE
Criar estrutura inicial do front-end

### ISSUE THREAD

**Issue by reebr** [dunossauro/curriculum-online#6]
Pode ser utilizado o [Parcel][1] para trabalhar com recursos mais modernos no front-end utilizando pouca configuração (pode ser feita uma comparação com o [webpack][6] para ver qual é melhor). Ele consegue lidar com [vários tipos de arquivos][2] *out-of-box*, sem a necessidade de arquivos extras de configurações específicas.

Algumas tecnologias que podem ser utilizadas a princípio:
- [PugJS][3]: para escrever o HTML de forma simples.
- [Sylus][4]: para criar as folhas de estilos de forma mais programática.
- [JS][5]: preferencialmente ES6 :shipit: 
- [NodeJS][7]
- [Yarn][8]

[1]:https://parceljs.org/
[2]:https://parceljs.org/assets.html#recursos-suportados-por-padr%C3%A3o
[3]:https://pugjs.org/
[4]:http://stylus-lang.com/
[5]:https://developer.mozilla.org/pt-BR/docs/Web/JavaScript
[6]:https://webpack.js.org/
[7]:https://nodejs.org/en/
[8]:https://yarnpkg.com/pt-BR/

**Comment by jlugao** [comment]
Eu ia sugerir um Vue.js com vuetify ou alguma outra lib que dê estilos prontos e seja fácil de usar.

**Comment by reebr** [comment]
Estilo pronto?

**Comment by jlugao** [comment]
Sim, não vejo problema em usar mais recursos para resolver um problema rápido e que permita mais gente prototipar rapidamente. Se a preocupação é não ter muito "bloat" podemos usar o bulma que é um framework bem menor.

Acho que seria legal começar já usando o Vue e criar a aplicação no estilo SPA, podemos usar editores inline e fazer algum tipo de visualização do currículo, se aproveitar de plugins drag-n-drop para mudar ordem das seções do currículo, e renderizar os templates de acordo com o dado. Mas também é o stack que estou acostumado para resolver o problema que estou imaginando.

Essa decisão na real depende de quais features teremos no frontend, talvez isso seja o mais importante de entender nesse momento. Que tal a gente esperar um pouco e discutir o que precisa ter para avançarmos na escolha do stack? O que vc está imaginando que teremos @reebr? O que vc acha que seria legal ter @dunossauro? 

ps. sobre o Pug eu sou bem contra, html é bem padrão e a maioria das pessoas sabe ler e utilizar, enquanto pug apenas algumas pessoas conhecem (e eu particularmente acho pouco legível).

**Comment by reebr** [comment]
Então, mas se a questão é ter legibilidade e ser fácil com que outras pessoas contribuam, colocar frameworks ao invés de escrever de forma "pura" não tornam as coisas mais difíceis? PUG é basicamente [html sem tags][3]; stylus não foge muito da forma como CSS é escrito, tirando que operacionalmente podem ser omitidos pontuações e chaves, e Javaccript é Javascript. 😄 

Conversando com o @dunossauro, a aplicação vai ser baseada no [resume-maker][1] que basicamente espera um [JSON][2] de entrada e gera o CV. Por isso disse que imagino algo simples, pelo menos a princípio.

PS: Sugeri o PUG não só pela simplicidade de escrever HTML mas também para lidar com a "componentização" que a galera faz dentro de arquivos JS hoje.

[1]:https://github.com/dunossauro/resume-maker
[2]:https://github.com/dunossauro/resume-maker/blob/master/examples/info.json
[3]:https://github.com/reebr/mock-ui/blob/master/src/template/download.pug

**Comment by jlugao** [comment]
Eu tinha escrito um textão justificando uma possível escolha de stack e a diferença de usar um bootstrap da vida e um pug, mas prefiro deixar essa conversa de lado para focarmos na questão mais importante:

Esse json pode ser gerado de várias formas diferentes, eu pensei que uma forma legal de trabalharmos isso seria com seções, define um título para a seção, tópicos (com título e conteúdo - que pode ser texto, uma escala numérica ou outros tipos a serem definidos) e algumas seções podem ter tópicos em que você só edita o conteúdo (tipo nome, telefone, email, etc...) talvez isso possa ser até definido pelo template em sí.

É importante entender o que a gente precisa fazer para decidir as ferramentas e aí discutir em cima dos requisitos. Pode detalhar um pouco melhor como você está imaginando que vai ser?
